c/tskit/trees.c

```diff
@@ -2076,6 +2076,32 @@ typedef struct {
     const tsk_id_t *index_tuples;
 } indexed_weight_stat_params_t;
 
+static int
+init_sample_weights(const tsk_treeseq_t *self, tsk_size_t num_sample_sets,
+    const tsk_size_t *sample_set_sizes, const tsk_id_t *sample_sets, double **weights)
+{
+    int ret = 0;
+    tsk_id_t u, sample_index;
+    tsk_size_t j, k, l;
+    double *weight_row;
```

Comment:
style nit: we'd usually put a blank line between the declarations and the function code

Reply:
Yeah, we do keep all the declarations at the start. The original reason was to support older C compilers, but now we're using C99 that doesn't apply any more. Now it's mostly for consistency, but there are still two good reasons for doing it this way:

1. It helps keep track of how big and complicated your function is (if you're declaring too many variables, the function is too big)
2. It helps keep a consistent pattern for memory management, and safety. All pointers are initialised to NULL before any code is run, and everything is ``safe_free'd`` at the end of the function.

Reply:
Makes sense, I've gone ahead moved all declarations to the start to be consistent with this style.

Reply:
Thanks :+1: